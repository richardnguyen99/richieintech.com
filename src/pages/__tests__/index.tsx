/**
 * Testing suite for Index (Home) page
 *
 * @author Richard Nguyen <richard.ng0616@gmail.com>
 */
import React from "react";
import { StaticQuery } from "gatsby";
import renderer from "react-test-renderer";

import { IndexQueryQuery } from "@generated/graphql";
import Index from "../index";

beforeEach(() => {
  (StaticQuery as jest.Mock<StaticQuery>).mockImplementation(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: "Richie in tech",
          titleTemplate: "%s | Richie in tech",
          description:
            "Official blog of Richie about techs, programing and coding.",
          url: "https://www.richieintech.com",
          twitterUsername: "@richardnguyenmh",
        },
      },
      placeholderImage: {
        childImageSharp: {
          fluid: {
            aspectRatio: 2,
            src: `test_image.jpg`,
            srcSet: `some srcSet`,
            srcSetWebp: `some srcSetWebp`,
            sizes: `(max-width: 600px) 100vw, 600px`,
            base64: `string_of_base64`,
          },
        },
      },
    })
  );
});

describe("Index page", () => {
  it("matches snapshot", () => {
    const data: IndexQueryQuery = {
      site: {
        siteMetadata: {
          defaultTitle: "Richie in tech",
          titleTemplate: "%s | Richie in tech",
          defaultDescription:
            "Official blog of Richie about techs, programing and coding.",
          siteUrl: "https://www.richieintech.com",
          twitterUsername: "@richardnguyenmh",
        },
      },
      placeholderImage: {
        childImageSharp: {
          fluid: {
            aspectRatio: 2,
            src: `test_image.jpg`,
            srcSet: `some srcSet`,
            srcSetWebp: `some srcSetWebp`,
            sizes: `(max-width: 600px) 100vw, 600px`,
            base64: `string_of_base64`,
          },
        },
      },
    };
    const tree = renderer.create(<Index data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
