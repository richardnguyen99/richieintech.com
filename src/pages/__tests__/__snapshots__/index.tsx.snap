// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
Array [
  <nav
    className="Navbar Navbar--light"
    id="main-header"
  >
    <div
      className="Container"
    >
      <a
        className="Navbar__brand"
        href="/"
      >
        <svg
          className="block"
          fill="none"
          height={50}
          viewBox="0 0 198 198"
          width={50}
        >
          <rect
            fill="#3C3C3C"
            height={140}
            rx={23}
            transform="rotate(-45 0 98.995)"
            width={140}
            y={98.9949}
          />
          <rect
            height={104.716}
            rx={14.5}
            stroke="#F5F5F5"
            strokeWidth={7}
            transform="rotate(-45 25.452 99.498)"
            width={104.716}
            x={25.4523}
            y={99.4975}
          />
          <path
            d="M78.176 117.776l4.928-.64V79.76l-4.672-.896v-4.416h9.856c.81 0 1.685-.021 2.624-.064a61.142 61.142 0 002.24-.128c2.304-.17 4.267-.256 5.888-.256 4.565 0 8.192.448 10.88 1.344 2.688.853 4.629 2.219 5.824 4.096 1.237 1.835 1.856 4.267 1.856 7.296 0 2.901-.768 5.61-2.304 8.128-1.536 2.517-3.797 4.245-6.784 5.184 1.963.725 3.605 2.091 4.928 4.096 1.365 1.963 2.773 4.48 4.224 7.552.853 1.792 1.536 3.136 2.048 4.032.512.853 1.045 1.344 1.6 1.472l1.856.576V122h-12.16c-.64 0-1.301-.683-1.984-2.048s-1.557-3.499-2.624-6.4c-1.323-3.541-2.475-6.272-3.456-8.192-.939-1.963-1.984-3.157-3.136-3.584h-5.76v15.36l5.696.64V122H78.176v-4.224zm20.8-20.16c2.219 0 3.947-.853 5.184-2.56 1.237-1.75 1.856-4.373 1.856-7.872 0-3.072-.619-5.312-1.856-6.72-1.195-1.45-3.221-2.176-6.08-2.176-1.536 0-2.88.107-4.032.32v18.816c.555.128 1.472.192 2.752.192h2.176z"
            fill="#F5F5F5"
          />
        </svg>
      </a>
      <ul
        className="Navbar__nav pl-0"
      >
        <li
          className="Navbar__item"
        >
          <a
            className="Navbar__link"
            href="/blog"
          >
            Tag
          </a>
        </li>
        <li
          className="Navbar__item"
        >
          <a
            className="Navbar__link"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          className="Navbar__item"
        >
          <a
            className="Navbar__link"
            href="/blog"
          >
            Guide
          </a>
        </li>
      </ul>
      <div
        className="Navbar__collapse"
      >
        <ul
          className="Navbar__nav ml-auto"
        >
          <li
            className="Navbar__item"
          >
            About
          </li>
          <li
            className="Navbar__item"
          >
            Hello
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div>
    <div>
      Hello, World!
    </div>
  </div>,
]
`;
